src/extra_files.c

```diff
@@ -63,15 +56,34 @@ static inline int bsearch_helper(const void *A, const void *B)
 
 static int record_filename(const char *name, const struct stat *stat __attribute__((unused)), int type, struct FTW *ftw __attribute__((unused)))
 {
-	for (size_t i = 0; i < sizeof(skip_dirs) / sizeof(skip_dirs[0]); i++) {
-		if (strncmp(name, skip_dirs[i].name, skip_dirs[i].len) == 0) {
+	/* Name as it would appear in manifest, f.i. /usr */
+	const char *relname = name + path_prefix_len - 1;
+
+	if (!relname[0] || !strcmp(relname, "/")) {
+		/* Do not record root while descending into it. */
+		return 0;
+	}
+
+	if (path_whitelist) {
+		int match_res;
+		if (type == FTW_D) {
+			/* include trailing slash for regex like ^/var/ */
```

Comment:
I still don't understand the motivation for this code.

You are adding a / to the end of directory names because you are asking the user to do the same, but what problem are you trying to solve?